File: patterns/js/pattern-20.js
/**

*        *
**      **
***    ***
****  ****
**********
****  ****
***    ***
**      **
*        *

 */

let line= '';
let N=5;
let end;
for(let i=1;i<=N;i++){
    for(let j=1;j<=i;j++){
        line+='*';
        end=j
    }
    let spaces = 2*(N-i);
    if(spaces){
        for(let j=0;j<spaces;j++){
            line+=' '
        }
    }
    for(let j=1;j<=i;j++){
        line+='*';
    }
    console.log(line);
    line='';
    end=''
}

for(let i=2;i<=N;i++){
    for(let j=i;j<=N;j++){
        line+=`*`;
        end=j
    }
    let spaces = 2*i-2;
    if(spaces){
        for(let j=0;j<spaces;j++){
            line+=' '
        }
    }
    
    for(let j=i;j<=N;j++){
        line+=`*`;
        end=''
    }
    console.log(line);
    line='';
    end=''
}

File: patterns/python/pattern20.py
'''
*        *
**      **
***    ***
****  ****
**********
****  ****
***    ***
**      **
*        *
'''

n=5
for i in range(n):
    for j in range(i+1):
        print('*',end='')
    spaces = 2*(n-1)-2*i
    for j in range(spaces):
        print(" ",end='')
    for j in range(i+1):
        print('*',end='')
    print()
n=n-1
for i in range(n):
    for j in range(n-i):
        print('*',end='')
    spaces = 2*(i+1)
    for j in range(spaces):
        print(' ',end='')
    for j in range(n-i):
        print('*',end='')
    print()